### ISSUE TITLE
[P2] Migrate send_coroutine fallback to use thread_pool

### ISSUE THREAD

**Issue by kcenon** [kcenon/network_system#274]
## Summary

Replace `std::thread().detach()` in `send_coroutine.cpp` fallback implementation with thread_pool task submission.

## Current Implementation

**File**: `src/internal/send_coroutine.cpp`

```cpp
// Lines 112-127 - Non-coroutine fallback
auto async_send_with_pipeline_no_co(...) -> std::future<std::error_code>
{
    auto promise = std::make_shared<std::promise<std::error_code>>();
    auto future_result = promise->get_future();
    
    auto future_processed = prepare_data_async(std::move(data), pl, use_compress, use_encrypt);
    
    // Problem: Creates detached thread
    std::thread([sock, promise, future = std::move(future_processed)]() mutable {
        try {
            auto processed_data = future.get();
            sock->async_send(std::move(processed_data),
                [promise](std::error_code ec, std::size_t) {
                    promise->set_value(ec);
                });
        }
        catch (const std::exception& e) {
            promise->set_value(std::make_error_code(std::errc::io_error));
        }
    }).detach();
    
    return future_result;
}
```

## Problem

- Creates detached threads for each send operation
- No control over thread lifecycle
- Potential resource exhaustion under high load

## Proposed Solution

Use `thread_integration_manager` to submit tasks to the shared thread pool:

```cpp
#include "kcenon/network/integration/thread_integration.h"

auto async_send_with_pipeline_no_co(...) -> std::future<std::error_code>
{
    auto promise = std::make_shared<std::promise<std::error_code>>();
    auto future_result = promise->get_future();
    
    auto future_processed = prepare_data_async(std::move(data), pl, use_compress, use_encrypt);
    
    // Use thread pool instead of detached thread
    integration::thread_integration_manager::instance().submit_task(
        [sock, promise, future = std::move(future_processed)]() mutable {
            try {
                auto processed_data = future.get();
                sock->async_send(std::move(processed_data),
                    [promise](std::error_code ec, std::size_t) {
                        promise->set_value(ec);
                    });
            }
            catch (const std::exception& e) {
                promise->set_value(std::make_error_code(std::errc::io_error));
            }
        });
    
    return future_result;
}
```

## thread_system Features to Use

- `thread_integration_manager`: Centralized thread pool access
- `thread_pool`: Task execution with managed workers

## Tasks

- [ ] Replace `std::thread().detach()` with `thread_integration_manager::submit_task()`
- [ ] Add error handling for pool submission failures
- [ ] Add unit tests
- [ ] Verify performance under high load

## Acceptance Criteria

- [ ] No `std::thread().detach()` in `send_coroutine.cpp`
- [ ] Tasks submitted to shared thread pool
- [ ] All existing tests pass
- [ ] No performance regression

## Related

- Epic: #271
- Depends on: #272
